### PR TITLE
TMA-775: platform agnostic lookup of label

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 1]
```diff
@@ -140,10 +140,11 @@ def self.verify_existing_users(filters, options = {})
 
     def self.create_label_cache(result, options = {})
       project = options[:project]
+      project_labels = project.labels
 
       result.reduce({}) do |a, e|
         e[:filters].map do |filter|
-          a[filter[:label]] = project.labels(filter[:label]) unless a.key?(filter[:label])
+          a[filter[:label]] = project_labels.find { |l| l.identifier == filter[:label]  } unless a.key?(filter[:label])
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Calls 'filter[:label]' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Contains iterators nested 3 deep (NestedIterators)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has the variable name 'l' (UncommunicativeVariableName)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Style/ExtraSpacing: Unnecessary spacing detected.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/659/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/454/ : FAILURE in 45s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/762/ : FAILURE in 2m 01s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/660/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/455/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/763/ : SUCCESS in 40s

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 2]
```diff
@@ -140,10 +140,11 @@ def self.verify_existing_users(filters, options = {})
 
     def self.create_label_cache(result, options = {})
       project = options[:project]
+      project_labels = project.labels
 
       result.reduce({}) do |a, e|
         e[:filters].map do |filter|
-          a[filter[:label]] = project.labels(filter[:label]) unless a.key?(filter[:label])
+          a[filter[:label]] = project_labels.find { |l| (l.identifier == filter[:label]) || (l.uri == filter[:label]) } unless a.key?(filter[:label])
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Calls 'filter[:label]' 4 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/661/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/456/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/764/ : SUCCESS in 33s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/253/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/457/ : SUCCESS in 2m 00s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/765/ : SUCCESS in 2m 01s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1169/ : FAILURE in 51m 27s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/927/ : SUCCESS in 34m 34s

**Comment by panjan** [comment]
The job gooddata-ruby-integration-tests-zuul/1169 failed in the clean-up phase. Merging as this is a hotfix.

**Pull request merged**